File: Dragon_Dungeons/Models/Npc.cs
namespace Dragon_Dungeons.Models;

public class Npc : RepoItem
{
  public string Name { get; set; }
  public string Picture { get; set; }
  public string Class { get; set; }
  public string Race { get; set; }
  public string CharacterId { get; set; }
  public string CampaignId { get; set; }
}

public class Player : Npc
{
  public string CreatorId { get; set; }
}

File: Dragon_Dungeons.client/src/models/Npc.js
import { RepoItem } from "./RepoItem.js"

export class Npc extends RepoItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.picture = data.picture
    this.class = data.class
    this.race = data.race
    this.characterId = data.characterId
    this.campaignId = data.campaignId
  }
}

export class Player extends Npc {
  constructor(data) {
    super(data)
    this.creatorId = data.creatorId
  }
}